[KOGITO-4336] PostgreSQL persistence support for Data Index (#3843)
kiegroup/kie-docs: doc-content/kogito-docs/src/main/asciidoc/openshift/chap-kogito-deploying-on-openshift.adoc

<Start here>
////

=== PostgreSQL persistence with the {PRODUCT} Data Index Service

> Data Index Service doesn't initialize its database schema automatically. To initialize the database schema you need to either enable automatic schema initialization using Hibernate or create the schema using DDL script.

Reviewer: ```suggestion
Data Index Service does not initialize its database schema automatically. To initialize the database schema, you need to either enable automatic schema initialization using Hibernate or create the schema using the DDL script.
```

> Automatic schema initialization using Hibernate can be enabled by application property `quarkus.hibernate-orm.database.generation` (see https://quarkus.io/guides/hibernate-orm#quarkus-hibernate-orm_quarkus.hibernate-orm.database.generation[Quarkus docs for database generation options] for possible values). The value can also be set as environment variable `QUARKUS_HIBERNATE_ORM_DATABASE_GENERATION`.

Reviewer: ```suggestion
Automatic schema initialization using Hibernate can be enabled using the `quarkus.hibernate-orm.database.generation` property. For possible values `quarkus.hibernate-orm.database.generation` property, see https://quarkus.io/guides/hibernate-orm#quarkus-hibernate-orm_quarkus.hibernate-orm.database.generation[Quarkus docs for database generation options]. The value can also be set as an environment variable `QUARKUS_HIBERNATE_ORM_DATABASE_GENERATION`.
```

> Example `DataIndex` resource with PostgreSQL persistence enabled (requires also Kafka KogitoInfra to be available)

Reviewer: ```suggestion
For example, the `DataIndex` resource with PostgreSQL persistence enabled, also requires Kafka KogitoInfra to be available:
```

Author: Well, the suggested modification is not much understandable for me.
I wanted to show user the possible CustomResource which can be used to create the Data index with PostgreSQL (the example is in the docs below). As a note I wanted to mention that Kafka KogitoInfra also needs to be available.

Reviewer: Oh okay, I've updated the suggestion above, please take a look

Author: Looks good, PR updated
